cloud/shared/bin/validate_cloud_provider

```diff
@@ -0,0 +1,13 @@
+#! /usr/bin/env bash
+
+readonly VALID_CLOUD_PROVIDERS=(azure)
```

Comment:
question (blocking): is this file going to remain the source of truth for what `CLOUD_PROVIDER` values are valid? I thought we were using an enum type variable definition here to enforce the accepted values:

https://github.com/seattle-uat/civiform/blob/715234ed2b9930f90edc84e3af855815903e244e/cloud/shared/variable_definitions.json#L6

Reply:
I thought so too, but I was writing/thinking about this it just seemed easier to have this one config var validated here and then load everything else after we've figured out which cloud provider we're working with. Otherwise we'll end up loading a bunch of the same code once before we've resolved the cloud provider and then a second time after.

If you'd rather take it the other direction I'm totally up for seeing what that could look like though.

Reply:
That makes sense. A thought I had was that we could validate all of the shared variables at the same time we validate the cloud provider, and then validate the provider-specific ones in the provider-specific deploy script.

Reply:
> A thought I had was that we could validate all of the shared variables at the same time we validate the cloud provider, and then validate the provider-specific ones in the provider-specific deploy script.

When you get into config validation, if that approach ends up looking better please feel free to restructure this a bit!